src/noble/hashes.ts

```diff
@@ -0,0 +1,20 @@
+import { sha256 } from '@noble/hashes/sha256'
+import { sha384, sha512 } from '@noble/hashes/sha512'
+import { CHash, wrapConstructor } from '@noble/hashes/utils'
+import { shake256 } from '@noble/hashes/sha3'
+import { HashID } from '../crypto.js'
+
+export const shake256_512 = wrapConstructor(() => shake256.create({ dkLen: 64 }))
+
+const HASHES: Record<HashID, CHash> = {
+    'SHA-256': sha256,
+    'SHA-384': sha384,
+    'SHA-512': sha512,
+    'SHAKE256': shake256_512
```

Comment:
Still tend to think SHAKE256-512 is more appropriate

Reply:
Given the hashes are all `(input: Uint8Array) => Uint8Array` conforming and openssl/noble uses 256 bits of output by default ?

Reply:
after a second thought, either name works for me.